src/Utilities/Compiler/Extensions/IEnumerableExtensions.cs

```diff
@@ -18,12 +18,17 @@ public static IEnumerable<T> Concat<T>(this IEnumerable<T> source, T value)
                 throw new ArgumentNullException(nameof(source));
             }
 
-            foreach (T v in source)
+            return ConcatImpl(source, value);
```

Comment:
With the previous implementation the null check wasn't performed before any actual iteration of the enumerable was performed.